### ISSUE TITLE
Support for adding Terraform Versions (Site Admin Functionality)

### ISSUE THREAD

**Issue by straubt1** [hashicorp/terraform-provider-tfe#135]
Looking for a resource to add new Terraform Versions as they become available.

Something like the following:

```hcl
resource "tfe_terraform_version" "12-20" {
  enabled  = true  #Make the Terraform version available to all users.
  beta     = false #Beta Terraform versions are only available to organizations that have been granted beta access.
  url      = "https://releases.hashicorp.com/terraform/0.12.0/terraform_0.12.0_linux_amd64.zip"
  checksum = "42ffd2db97853d5249621d071f4babeed8f5fdba40e3685e6c1013b9b7b25830" #The SHA256 checksum of the downloaded Terraform version will be verified. The checksum can be found on the download page.
}
```

Ideally this would act as an upsert when dealing with an existing Terraform Version.
TFE currently ships with over 100 Terraform Versions.

**Comment by straubt1** [comment]
Any thoughts on this?

**Comment by chrisarcand** [comment]
I think this would be awesome, and am eager to do it.

We (the team at HashiCorp now maintaining this provider) are currently planning on how we want to incorporate Site Administration functionality within this provider, which is why this and several other open PRs/Issues about Site Admin stuff remain open at the moment. The Site Admin API is a separate API from the API which this provider utilizes to-date, so we're taking extra time to be plan on how that introduction plays out (because once one Site Admin features is added, there's certainly no reason any or most of them couldn't be!)

It's on our immediate roadmap and we hope to incorporate this and many others like it, soon. Thanks for everyone's patience!

**Comment by chrisarcand** [comment]
An update, for interested parties!

We're close to merging support for this in the Go client, [here](https://github.com/hashicorp/go-tfe/pull/186).

**Comment by dkulchinsky** [comment]
Hey @chrisarcand 👋🏼 

the change in go-tfe was merged back in March, any plans/timeline to implement this functionality in the provider now that it's unblocked in the library?

**Comment by chrisarcand** [comment]
Resolved in #400, and will be included in the next release.
